pkgs/development/libraries/libibverbs/default.nix

```diff
@@ -1,17 +1,78 @@
 { stdenv, fetchurl }:
 
-stdenv.mkDerivation rec {
-  name = "libibverbs-1.1.8";
+let
+  
+  verbs = rec {
+      version = "1.1.8";
+      name = "libibverbs-${version}";
+      url = "http://downloads.openfabrics.org/verbs/${name}.tar.gz";
+      sha256 = "13w2j5lrrqxxxvhpxbqb70x7wy0h8g329inzgfrvqv8ykrknwxkw";
+  };
 
-  src = fetchurl {
-    url = "https://www.openfabrics.org/downloads/verbs/${name}.tar.gz";
-    sha256 = "13w2j5lrrqxxxvhpxbqb70x7wy0h8g329inzgfrvqv8ykrknwxkw";
+  drivers = {
+     libmlx4 = rec { 
+                     version="1.0.6";
+                     name="libmlx4-${version}"; 
+                     url="http://downloads.openfabrics.org/mlx4/${name}.tar.gz";
+                     sha256="f680ecbb60b01ad893490c158b4ce8028a3014bb8194c2754df508d53aa848a8";
+                   };
+     libmthca = rec { 
+                      version="1.0.6"; 
+                      name="libmthca-${version}"; 
+                      url="http://downloads.openfabrics.org/mthca/${name}.tar.gz";
+                      sha256="cc8ea3091135d68233d53004e82b5b510009c821820494a3624e89e0bdfc855c";
+                    };
   };
 
+in stdenv.mkDerivation rec {
+
+  version = verbs.version;
+  name = verbs.name;
+
+  srcs = [
+    (fetchurl {
+      url = verbs.url;
+      sha256 = verbs.sha256;
+    })
+
+    (fetchurl {
+      url = drivers.libmlx4.url;
+      sha256 = drivers.libmlx4.sha256;
+    })
+
+    (fetchurl {
+      url = drivers.libmthca.url;
+      sha256 = drivers.libmthca.sha256;
+    })
+ 
+  ];
+
+  sourceRoot=name;
+
+  # Install userspace drivers
+  postInstall=''
```

Comment:
I think your solution, while a bit ugly, is workable. It is a hairy problem. If what you got works, I'm not against taking it as-is, but I'll leave this open a bit longer, just in case someone has a good alternative. Let me know if it ends up lingering for too long (I realize it's been a while already).